as/thrift.js

```diff
@@ -41,17 +41,14 @@ function TChannelAsThrift(opts) {
 
     var self = this;
 
-    assert(opts && typeof opts.source === 'string',
-        'must pass source as an argument');
-
     self.thriftSource = opts.source;
-    self.thriftFileName = opts.thriftFileName || 'service.thrift';
+    self.thriftFileName = path.basename(opts.entryPoint || 'service.thrift');
     self.spec = new thriftrw.Thrift({
         entryPoint: opts.entryPoint,
         idls: opts.idls,
         source: self.thriftSource,
         strict: opts.strict,
-        allowFilesystemAccess: opts.allowFilesystemAccess,
+        allowFilesystemAccess: true,
```

Comment:
Let’s assume not for now and provide the opposite option later if needed. The only case we would disallow filesystem access for includes is when we construct Thrift from Meta::thriftIDLs, which we only intend to do in tools like tcurl/tcap.